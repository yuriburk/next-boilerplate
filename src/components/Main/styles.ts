import styled from 'styled-components'

export const Container = styled.main`
  display: flex;
  align-items: center;
  justify-content: center;
  flex: 1;
  width: 100%;
  height: 100%;
  color: #ffffff;
  background-color: #05092b;
`
export const Title = styled.h1`
  font-size: 2rem;
  font-weight: 400;
`
