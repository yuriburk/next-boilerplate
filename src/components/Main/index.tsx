import * as SC from './styles'

const Main = () => (
  <SC.Container>
    <SC.Title>Burk.dev</SC.Title>
  </SC.Container>
)

export default Main
