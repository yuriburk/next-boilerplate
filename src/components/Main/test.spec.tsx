import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should be able to render the heading', () => {
    const { container } = render(<Main />)

    expect(screen.getByRole('heading', { name: /burk.dev/i })).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should be able to render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#05092b' })
  })
})
