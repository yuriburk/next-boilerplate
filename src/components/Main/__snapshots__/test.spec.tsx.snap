// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should be able to render the heading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  width: 100%;
  height: 100%;
  color: #ffffff;
  background-color: #05092b;
}

.c1 {
  font-size: 2rem;
  font-weight: 400;
}

<div>
  <main
    class="c0"
  >
    <h1
      class="c1"
    >
      Burk.dev
    </h1>
  </main>
</div>
`
